### PR TITLE
Adding resources for dynamic type ahead UI

### PR DESCRIPTION
**Pull request by dipja**
# Feature - [Android][Second PR] Dynamic Type Ahead
This is the second PR covers code changes to add drawables as UI resources in dynamic type ahead new screen. Also, provides support for custom icons from host.

Feature Details: https://github.com/microsoft/AdaptiveCards/blob/dipja/dynamic-type-ahead-doc-android/specs/DesignDiscussions/DynamicTypeAhead.md

Design Doc: https://github.com/microsoft/AdaptiveCards/blob/dipja/dynamic-type-ahead-doc-android/specs/DesignDiscussions/DynamicTypeAheadAndroid.md
 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://api.github.com/repos/microsoft/AdaptiveCards/pulls/8273)
 
 
 Type ahead state icons
 
 Start Searching State Icon
Screenshot 2023-02-20 at 2 58 12 PM

Error/NoResult State Icon
Screenshot 2023-02-20 at 2 58 30 PM

Select/Tick Icon
Screenshot 2023-02-20 at 2 58 20 PM

Chevon Icon
Screenshot 2023-02-20 at 2 57 36 PM

Search Icon
Screenshot 2023-02-20 at 2 56 55 PM

Back Icon
Screenshot 2023-02-20 at 2 57 56 PM

Cross Button
Screenshot 2023-02-20 at 2 57 49 PM

### CODE REVIEW

**Comment by almedina-ms** [comment]
As this PR is focusing on rendering and translating a design into code it would be great to have images or videos that show how the element is looking like at this point

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/ITypeAheadCustomParams.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
```
Comment (almedina-ms) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/ITypeAheadCustomParams.java:2]: please migrate to kotlin

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/http/HttpRequestResult.java [hunk 2]
```diff
@@ -26,15 +32,16 @@ public RESULT getResult()
...
     public boolean isSuccessful()
     {
         return m_success;
     }
 
-    private Exception m_excep = null;
+    private Exception m_exception = null;
     private RESULT m_result = null;
     private boolean m_success = true;
+    private String error_message = null;
```
Comment (almedina-ms) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/http/HttpRequestResult.java:46]: rename to m_errorMessage

Comment (almedina-ms) [reply source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/http/HttpRequestResult.java:46]: is this property missing a public getter?

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/registration/CardRendererRegistration.java [hunk 3]
```diff
@@ -256,6 +257,14 @@ public IActionLayoutRenderer getOverflowActionLayoutRenderer()
         return m_overflowActionLayoutRenderer;
     }
 
+    public void registerTypeAheadParams(ITypeAheadCustomParams typeAheadCustomParams) {
```
Comment (almedina-ms) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/registration/CardRendererRegistration.java:260]: format to match the remainder of the document

File: source/android/adaptivecards/src/main/res/drawable/adaptive_card_ic_baseline_search_24.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (almedina-ms) [inline source/android/adaptivecards/src/main/res/drawable/adaptive_card_ic_baseline_search_24.xml:1]: if possible add copyright headers

File: source/android/adaptivecards/src/main/res/drawable/adaptive_card_ic_baseline_search_24.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    android:width="24dp"
```
Comment (almedina-ms) [inline source/android/adaptivecards/src/main/res/drawable/adaptive_card_ic_baseline_search_24.xml:2]: in general, when adding visual elements, please share a screenshot of how the rendered visual looks like

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/registration/CardRendererRegistration.java (hunk 3 above)
Comment (dipja) [reply source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/registration/CardRendererRegistration.java]: done

File: source/android/adaptivecards/src/main/res/drawable/adaptive_card_ic_baseline_search_24.xml (hunk 4 above)
Comment (dipja) [reply source/android/adaptivecards/src/main/res/drawable/adaptive_card_ic_baseline_search_24.xml:4]: done

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/http/HttpRequestResult.java (hunk 2 above)
Comment (dipja) [reply source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/http/HttpRequestResult.java]: Thanks for pointing out the missing getter. Added!

**Review by almedina-ms — APPROVED** [review]

**Pull request merged**